torch/utils/tensorboard/summary.py

```diff
@@ -73,6 +72,95 @@ def _draw_single_box(image, xmin, ymin, xmax, ymax, display_str, color='black',
 
 
+def hparams(hparam_dict=None, metric_dict=None):
+    """Outputs three `Summary` protocol buffers needed by hparams plugin.
+    `Experiment` keeps the metadata of an experiment, such as the name of the
+      hyperparameters and the name of the metrics.
+    `SessionStartInfo` keeps key-value pairs of the hyperparameters
+    `SessionEndInfo` describes status of the experiment e.g. STATUS_SUCCESS
+
+    Args:
+      hparam_dict: A dictionary that contains names of the hyperparameters
+        and their values.
+      metric_dict: A dictionary that contains names of the metrics
+        and their values.
+
+    Returns:
+      The `Summary` protobufs for Experiment, SessionStartInfo and
+        SessionEndInfo
+    """
+    from tensorboard.plugins.hparams.api_pb2 import (
+        Experiment, HParamInfo, MetricInfo, MetricName, Status
+    )
+    from tensorboard.plugins.hparams.metadata import (
+        PLUGIN_NAME,
+        PLUGIN_DATA_VERSION,
+        EXPERIMENT_TAG,
+        SESSION_START_INFO_TAG,
+        SESSION_END_INFO_TAG
+    )
+    from tensorboard.plugins.hparams.plugin_data_pb2 import (
+        HParamsPluginData, SessionEndInfo, SessionStartInfo
+    )
+
+    # TODO: expose other parameters in the future.
+    # hp = HParamInfo(name='lr',display_name='learning rate',
+    # type=DataType.DATA_TYPE_FLOAT64, domain_interval=Interval(min_value=10,
+    # max_value=100))
+    # mt = MetricInfo(name=MetricName(tag='accuracy'), display_name='accuracy',
+    # description='', dataset_type=DatasetType.DATASET_VALIDATION)
+    # exp = Experiment(name='123', description='456', time_created_secs=100.0,
+    # hparam_infos=[hp], metric_infos=[mt], user='tw')
+
+    if not isinstance(hparam_dict, dict):
+        logging.warning('parameter: hparam_dict should be a dictionary, nothing logged.')
+        raise TypeError('parameter: hparam_dict should be a dictionary, nothing logged.')
+    if not isinstance(metric_dict, dict):
+        logging.warning('parameter: metric_dict should be a dictionary, nothing logged.')
+        raise TypeError('parameter: metric_dict should be a dictionary, nothing logged.')
+
+    hps = [HParamInfo(name=k) for k in hparam_dict.keys()]
+    mts = [MetricInfo(name=MetricName(tag=k)) for k in metric_dict.keys()]
+
+    exp = Experiment(hparam_infos=hps, metric_infos=mts)
+
+    content = HParamsPluginData(experiment=exp, version=PLUGIN_DATA_VERSION)
+    smd = SummaryMetadata(
+        plugin_data=SummaryMetadata.PluginData(plugin_name=PLUGIN_NAME,
+            content=content.SerializeToString()))
+    exp = Summary(value=[Summary.Value(tag=EXPERIMENT_TAG, metadata=smd)])
+
+    ssi = SessionStartInfo()
+    for k, v in hparam_dict.items():
+        if isinstance(v, str):
+            ssi.hparams[k].string_value = v
+            continue
+
+        if isinstance(v, bool):
+            ssi.hparams[k].bool_value = v
+            continue
+
+        if not isinstance(v, int) or not isinstance(v, float):
```

Comment:
Looks like straight int or float won't be set? Do you mean to have

```
            v = make_np(v)
        ssi.hparams[k].number_value = v
```
with the assignment outside the if block?